src/Architecture.cpp

```diff
@@ -185,21 +185,34 @@ Architecture::Architecture(const std::uint16_t nQ, const CouplingMap& cm,
 }
 
 void Architecture::createDistanceTable() {
-  isBidirectional = true;
+  isBidirectional  = true;
+  isUnidirectional = true;
   Matrix edgeWeights(nqubits, std::vector<double>(
                                   nqubits, std::numeric_limits<double>::max()));
   for (const auto& edge : couplingMap) {
     if (couplingMap.find({edge.second, edge.first}) == couplingMap.end()) {
+      // unidirectional edge
       isBidirectional                            = false;
       edgeWeights.at(edge.second).at(edge.first) = COST_UNIDIRECTIONAL_SWAP;
       edgeWeights.at(edge.first).at(edge.second) = COST_UNIDIRECTIONAL_SWAP;
     } else {
+      // bidirectional edge
+      isUnidirectional                           = false;
       edgeWeights.at(edge.first).at(edge.second) = COST_BIDIRECTIONAL_SWAP;
     }
   }
 
-  Dijkstra::buildTable(nqubits, couplingMap, distanceTable, edgeWeights,
-                       COST_DIRECTION_REVERSE, true);
+  Matrix simpleDistanceTable{};
+  Dijkstra::buildTable(couplingMap, simpleDistanceTable, edgeWeights);
+  Dijkstra::buildSingleEdgeSkipTable(simpleDistanceTable, couplingMap, 0.,
+                                     distanceTable);
+  if (bidirectional()) {
+    distanceTableReversals = distanceTable;
+  } else {
+    Dijkstra::buildSingleEdgeSkipTable(simpleDistanceTable, couplingMap,
+                                       COST_DIRECTION_REVERSE,
+                                       distanceTableReversals);
+  }
 }
```

Comment:
This has gotten quite a bit more complicated then before. Wouldn't it make sense to move this to where the Dijkstra routines are located so that it only becomes a single call here?
Also, I am still a bit puzzled by the need for the additional table, but I guess that will resolve after you had time to comment.

Reply:
Actually, now that you mention it, this is the case for `buildEdgeSkipTable`, which could directly call `buildTable` for its first dimension. I will quickly implement that and push it to this PR.

However, in the above case with `buildSingleEdgeSkipTable` we potentially need to build multiple different edge skip tables with different reversal costs. Currently, it is just 1 or 2 tables, which could just be communicated to the function via parameters, though even this would already unnecessarily increase coupling between those 2 classes in my opinion.
However, going forward with arbitrary 2q gates it will quite likely become necessary to build edge skip tables for even more different reversal costs and potentially do that even on the fly later on. So I think it is better to keep `buildTable` and `buildSingleEdgeSkipTable` separate from each other.

Reply:
Alright. Let's keep them separate for now. I guess the 2q-gates feature will show if this can be simplified or becomes even more complicated.